<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Zheng Li" />

<meta name="date" content="2022-06-12" />

<title>DLPFC data by 10x Visium (Maynard et al., 2021)</title>

<script src="site_libs/header-attrs-2.12.1/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">BASS</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Introduction</a>
</li>
<li>
  <a href="about.html">Software tutorial</a>
</li>
<li>
  <a href="simu.html">Simulation</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Real data analyses
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="STARmap.html">STARmap (Wang et al., 2018)</a>
    </li>
    <li>
      <a href="MERFISH.html">MERFISH (Moffitt et al., 2018)</a>
    </li>
    <li>
      <a href="DLPFC.html">DLPFC (Maynard et al., 2021)</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/zhengli09//BASS-analysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">DLPFC data by 10x Visium (Maynard et al.,
2021)</h1>
<h4 class="author">Zheng Li</h4>
<h4 class="date">2022-06-12</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-06-12
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>BASS-analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed0code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(0)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed0code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(0)</code> was run prior to running the
code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomzhengli09BASSanalysistree829d10c5addeb411ad5deb4dea9f3ff3e8d79d97targetblank829d10ca">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/zhengli09//BASS-analysis/tree/829d10c5addeb411ad5deb4dea9f3ff3e8d79d97" target="_blank">829d10c</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomzhengli09BASSanalysistree829d10c5addeb411ad5deb4dea9f3ff3e8d79d97targetblank829d10ca"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/zhengli09//BASS-analysis/tree/829d10c5addeb411ad5deb4dea9f3ff3e8d79d97" target="_blank">829d10c</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
working directory clean
</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/DLPFC.Rmd</code>) and HTML
(<code>docs/DLPFC.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/zhengli09//BASS-analysis/blob/829d10c5addeb411ad5deb4dea9f3ff3e8d79d97/analysis/DLPFC.Rmd" target="_blank">829d10c</a>
</td>
<td>
zhengli09
</td>
<td>
2022-06-12
</td>
<td>
Update real data analysis
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/zhengli09//BASS-analysis/blob/2f2db0dfe7a9811418e90116f425b3606109274a/analysis/DLPFC.Rmd" target="_blank">2f2db0d</a>
</td>
<td>
zhengli09
</td>
<td>
2022-03-05
</td>
<td>
Separate spatialLIBD data into three parts, each corresponding to an
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/zhengli09//BASS-analysis/blob/4b90802ca87082e56d533d41faf4d1bd9c16863a/analysis/DLPFC.Rmd" target="_blank">4b90802</a>
</td>
<td>
zhengli09
</td>
<td>
2022-03-05
</td>
<td>
Add DLPFC analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/zhengli09//BASS-analysis/b6853e0921de632124597c6e6731eacaea5aa8a8/docs/DLPFC.html" target="_blank">b6853e0</a>
</td>
<td>
zhengli09
</td>
<td>
2022-03-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/zhengli09//BASS-analysis/48fb04e74c197f5c419cb2d3100363d09430d74b/docs/DLPFC.html" target="_blank">48fb04e</a>
</td>
<td>
zhengli09
</td>
<td>
2022-03-05
</td>
<td>
Build site.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>Here, we apply BASS to analyze the human dorsolateral prefrontal
cortex (DLPFC) data by 10x Visium from <a
href="https://www.nature.com/articles/s41593-020-00787-0">Maynard et
al., 2021</a>. DLPFC data contains expression values of 33,538 genes
measured on two pairs of tissue sections from three independent
neurotypical adult donors. Each pair consisted of two directly adjacent,
10 <span class="math inline">\(\mu m\)</span> serial tissue sections
with the second pair located 300 <span class="math inline">\(\mu
m\)</span> posterior to the first, resulting in a total of 12 tissue
sections. The original data can be downloaded from <a
href="https://github.com/LieberInstitute/spatialLIBD">here</a>. We
excluded spots that were not mapped to the tissue region in the
histology image and retained a total of 33,538 genes measured on 4,226
(151507), 4,384 (151508), 4,789 (151509), 4,634 (151510), 3,661
(151669), 3,498 (151670), 4,110 (151671), 4,015 (151672), 3,639
(151673), 3,673 (151674), 3,592 (151675), and 3,460 (151676) spots along
with their spatial locations for further analysis. The processed data
can be found at the <a
href="https://github.com/zhengli09/BASS-Analysis/tree/master/data">data
directory</a>. We focused our analysis only on spatial domain detection
because the clustering of spatial spots no longer has the cell type
interpretation. For single-sample analysis, we analyzed each of the 12
tissue sections separately. For multi-sample analysis, we jointly
analyzed four tissue sections from each adult donor because they contain
similar tissue structures.</p>
</div>
<div id="load-data-and-set-hyper-parameters" class="section level1">
<h1>Load data and set hyper-parameters</h1>
<pre class="r"><code>library(BASS)
library(Matrix)
# cntm: a list of expression count matrices for 4 tissue sections
#       (151507-151510) from the same individual
# infom: a list of manually annotated labels of seven laminar 
#        clusters by the original study for 4 tissue sections
#        (151507-151510) from the same individual
# xym: a list of spatial coordinates for 4 tissue sections
#      (151507-151510) from the same individual 
load(&quot;data/spatialLIBD_p1.RData&quot;)

# hyper-parameters
# We set the number of cell types to a relatively large
# number (20) to capture the expression heterogeneity.
C &lt;- 20
# number of spatial domains
R &lt;- 7</code></pre>
</div>
<div id="run-bass" class="section level1">
<h1>Run BASS</h1>
<pre class="r"><code>set.seed(0)
# Set up BASS object
BASS &lt;- createBASSObject(cntm, xym, C = C, R = R,
  beta_method = &quot;SW&quot;, init_method = &quot;mclust&quot;, 
  nsample = 10000)</code></pre>
<pre><code>***************************************
  INPUT INFO:
    - Number of tissue sections: 4 
    - Number of cells/spots: 4226 4384 4789 4634 
    - Number of genes: 33538 
    - Potts interaction parameter estimation method: SW 
      - Estimate Potts interaction parameter with SW algorithm
  To list all hyper-parameters, Type listAllHyper(BASS_object)
***************************************</code></pre>
<pre class="r"><code># Data pre-processing:
# 1.Library size normalization followed with a log2 transformation
# 2.Select top 3000 spatially expressed genes with SPARK-X
# 3.Dimension reduction with PCA
BASS &lt;- BASS.preprocess(BASS, doLogNormalize = TRUE,
  geneSelect = &quot;sparkx&quot;, nSE = 3000, doPCA = TRUE, 
  scaleFeature = FALSE, nPC = 20)</code></pre>
<pre><code>***** Log-normalize gene expression data *****
***** Select spatially expressed genes with sparkx *****
***** Exclude genes with 0 expression *****
***** Reduce data dimension with PCA *****
***** Correct batch effect with Harmony *****</code></pre>
<pre class="r"><code># Run BASS algorithm
BASS &lt;- BASS.run(BASS)</code></pre>
<pre class="r"><code># post-process posterior samples:
# 1.Adjust for label switching with the ECR-1 algorithm
# 2.Summarize the posterior samples to obtain the spatial domain labels
BASS &lt;- BASS.postprocess(BASS)</code></pre>
<pre><code>Post-processing...
done</code></pre>
<pre class="r"><code>zlabels &lt;- BASS@results$z # spatial domain labels</code></pre>
</div>
<div id="visualization" class="section level1">
<h1>Visualization</h1>
<p>You can refer to <a
href="https://github.com/zhengli09/BASS-Analysis/tree/master/code/viz.R">visualization</a>
for some useful plotting functions or you can write your own code for
plotting.</p>
<pre class="r"><code>library(ggplot2)
source(&quot;code/viz.R&quot;)</code></pre>
<pre class="r"><code>zlabels &lt;- lapply(zlabels, function(zlabels.l){
  zlabels.l &lt;- factor(zlabels.l)
  levels(zlabels.l) &lt;- c(&quot;L3&quot;, &quot;L1&quot;, &quot;L5&quot;, &quot;L2&quot;, &quot;L4&quot;, &quot;WM&quot;, &quot;L6&quot;)
  zlabels.l &lt;- factor(zlabels.l, levels = c(
    &quot;L1&quot;, &quot;L2&quot;, &quot;L3&quot;, &quot;L4&quot;, &quot;L5&quot;, &quot;L6&quot;, &quot;WM&quot;
  ))
})
cols &lt;- c(&quot;#264653&quot;, &quot;#287271&quot;, &quot;#2a9d8f&quot;, &quot;#8ab17d&quot;, 
  &quot;#e9c46a&quot;, &quot;#f4a261&quot;, &quot;#e76f51&quot;)
smps &lt;- c(&quot;151507&quot;, &quot;151508&quot;, &quot;151509&quot;, &quot;151510&quot;)
plotClustersFacet(xym, zlabels, smps, size = 1) +
  scale_color_manual(&quot;Domain&quot;, values = cols) +
  guides(color = guide_legend(nrow = 1, 
    override.aes = list(size = 2)))</code></pre>
<p><img src="figure/DLPFC.Rmd/unnamed-chunk-5-1.png" width="1152" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.5 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas/libblas.so.3
LAPACK: /usr/lib/x86_64-linux-gnu/libopenblasp-r0.2.20.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ggplot2_3.3.5   Matrix_1.4-1    BASS_1.1.0      GIGrvg_0.5     
[5] workflowr_1.7.0

loaded via a namespace (and not attached):
  [1] bitops_1.0-7                matrixStats_0.61.0         
  [3] fs_1.5.2                    doParallel_1.0.15          
  [5] httr_1.4.2                  rprojroot_2.0.2            
  [7] GenomeInfoDb_1.24.2         tools_4.2.0                
  [9] bslib_0.3.1                 utf8_1.2.2                 
 [11] R6_2.5.1                    irlba_2.3.3                
 [13] matlab_1.0.2                vipor_0.4.5                
 [15] DBI_1.1.1                   BiocGenerics_0.38.0        
 [17] colorspace_2.0-3            withr_2.4.3                
 [19] label.switching_1.8         tidyselect_1.1.2           
 [21] gridExtra_2.3               processx_3.5.2             
 [23] compiler_4.2.0              git2r_0.28.0               
 [25] cli_3.2.0                   Biobase_2.48.0             
 [27] SPARK_1.1.1                 BiocNeighbors_1.6.0        
 [29] DelayedArray_0.18.0         labeling_0.4.2             
 [31] sass_0.4.1                  scales_1.1.1               
 [33] callr_3.7.0                 stringr_1.4.0              
 [35] digest_0.6.29               rmarkdown_2.12.1           
 [37] harmony_0.1.0               XVector_0.32.0             
 [39] scater_1.16.2               pkgconfig_2.0.3            
 [41] htmltools_0.5.2             sparseMatrixStats_1.8.0    
 [43] MatrixGenerics_1.4.3        highr_0.9                  
 [45] fastmap_1.1.0               rlang_1.0.1                
 [47] rstudioapi_0.13             DelayedMatrixStats_1.14.3  
 [49] farver_2.1.0                jquerylib_0.1.4            
 [51] generics_0.1.2              combinat_0.0-8             
 [53] jsonlite_1.8.0              mclust_5.4.9               
 [55] BiocParallel_1.22.0         dplyr_1.0.8                
 [57] RCurl_1.98-1.5              magrittr_2.0.2             
 [59] BiocSingular_1.4.0          GenomeInfoDbData_1.2.6     
 [61] Rcpp_1.0.8.3                ggbeeswarm_0.6.0           
 [63] munsell_0.5.0               S4Vectors_0.30.2           
 [65] fansi_1.0.2                 viridis_0.5.1              
 [67] lifecycle_1.0.1             stringi_1.7.6              
 [69] whisker_0.4                 yaml_2.3.5                 
 [71] CompQuadForm_1.4.3          SummarizedExperiment_1.22.0
 [73] zlibbioc_1.34.0             grid_4.2.0                 
 [75] blob_1.2.1                  parallel_4.2.0             
 [77] promises_1.1.1              crayon_1.5.0               
 [79] lattice_0.20-45             cowplot_1.1.1              
 [81] knitr_1.37                  ps_1.6.0                   
 [83] pillar_1.7.0                GenomicRanges_1.44.0       
 [85] lpSolve_5.6.15              codetools_0.2-18           
 [87] stats4_4.2.0                glue_1.6.2                 
 [89] evaluate_0.15               getPass_0.2-2              
 [91] foreach_1.5.0               vctrs_0.3.8                
 [93] httpuv_1.5.4                tidyr_1.1.1                
 [95] gtable_0.3.0                purrr_0.3.4                
 [97] assertthat_0.2.1            xfun_0.29                  
 [99] rsvd_1.0.3                  pracma_2.2.9               
[101] later_1.1.0.1               viridisLite_0.4.0          
[103] SingleCellExperiment_1.14.1 tibble_3.1.6               
[105] iterators_1.0.13            beeswarm_0.4.0             
[107] IRanges_2.26.0              ellipsis_0.3.2             </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
