<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Zheng Li" />

<meta name="date" content="2022-06-08" />

<title>Simulation</title>

<script src="site_libs/header-attrs-2.12.1/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">BASS</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Introduction</a>
</li>
<li>
  <a href="about.html">Software tutorial</a>
</li>
<li>
  <a href="simu.html">Simulation</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Real data analyses
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="STARmap.html">STARmap (Wang et al., 2018)</a>
    </li>
    <li>
      <a href="MERFISH.html">MERFISH (Moffitt et al., 2018)</a>
    </li>
    <li>
      <a href="DLPFC.html">DLPFC (Maynard et al., 2021)</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/zhengli09//BASS-analysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Simulation</h1>
<h4 class="author">Zheng Li</h4>
<h4 class="date">2022-06-08</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-06-08
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>BASS-analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed0code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(0)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed0code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(0)</code> was run prior to running the
code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomzhengli09BASSanalysistree3d28aa4d1d2f64a5804526198dfda5509a11318dtargetblank3d28aa4a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/zhengli09//BASS-analysis/tree/3d28aa4d1d2f64a5804526198dfda5509a11318d" target="_blank">3d28aa4</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomzhengli09BASSanalysistree3d28aa4d1d2f64a5804526198dfda5509a11318dtargetblank3d28aa4a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/zhengli09//BASS-analysis/tree/3d28aa4d1d2f64a5804526198dfda5509a11318d" target="_blank">3d28aa4</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Unstaged changes:
    Modified:   data/MERFISH_Animal1.RData
    Modified:   data/starmap_mpfc.RData

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/simu.Rmd</code>) and HTML
(<code>docs/simu.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/zhengli09//BASS-analysis/blob/3d28aa4d1d2f64a5804526198dfda5509a11318d/analysis/simu.Rmd" target="_blank">3d28aa4</a>
</td>
<td>
zhengli09
</td>
<td>
2022-06-08
</td>
<td>
Fix format
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/zhengli09//BASS-analysis/8b169e7cf961f4b59d9fa4803f7ca92946ad80a2/docs/simu.html" target="_blank">8b169e7</a>
</td>
<td>
zhengli09
</td>
<td>
2022-06-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/zhengli09//BASS-analysis/blob/90ee3a5f33e98e4e95a101aea7a1c0de663c6ce7/analysis/simu.Rmd" target="_blank">90ee3a5</a>
</td>
<td>
zhengli09
</td>
<td>
2022-06-08
</td>
<td>
Update the simulation section
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/zhengli09//BASS-analysis/5c699fe7266333d0d9b85de0afd9e0e2de2a4afc/docs/simu.html" target="_blank">5c699fe</a>
</td>
<td>
zhengli09
</td>
<td>
2022-03-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/zhengli09//BASS-analysis/blob/b835db9c76c85672873e7da97cb9ae0f1bd166c7/analysis/simu.Rmd" target="_blank">b835db9</a>
</td>
<td>
zhengli09
</td>
<td>
2022-03-10
</td>
<td>
Add figure of additional tissue sections
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/zhengli09//BASS-analysis/182beebb01b81150712c13602b2446a0c4405c44/docs/simu.html" target="_blank">182beeb</a>
</td>
<td>
zhengli09
</td>
<td>
2022-03-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/zhengli09//BASS-analysis/blob/34e84d6065c8b15a6d96cb444f9142b35d6667a3/analysis/simu.Rmd" target="_blank">34e84d6</a>
</td>
<td>
zhengli09
</td>
<td>
2022-03-10
</td>
<td>
Add software tutorial and multi-sample simulation
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/zhengli09//BASS-analysis/a27dd42cad44fadeffe07751ac770d859d28a83f/docs/simu.html" target="_blank">a27dd42</a>
</td>
<td>
zhengli09
</td>
<td>
2022-02-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/zhengli09//BASS-analysis/blob/ada7b84d38930e62d40ec33e901390ceb4a2b81f/analysis/simu.Rmd" target="_blank">ada7b84</a>
</td>
<td>
zhengli09
</td>
<td>
2022-02-27
</td>
<td>
Modify links
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/zhengli09//BASS-analysis/9a685114f1f29bb007c27088adaf8ce2582c56a2/docs/simu.html" target="_blank">9a68511</a>
</td>
<td>
zhengli09
</td>
<td>
2022-02-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/zhengli09//BASS-analysis/blob/4ec8df1262087c7c78f682e56014a7087a756d4f/analysis/simu.Rmd" target="_blank">4ec8df1</a>
</td>
<td>
zhengli09
</td>
<td>
2022-02-27
</td>
<td>
Publish the introduction, simulation analysis and STARmap analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>This page provides code to reproduce all the simulation results for
BASS. Hopefully, this can also serve as a resource for building up your
simulation studies and evaluate your statistical methods.</p>
</div>
<div id="main-simulation-study" class="section level1">
<h1>Main simulation study</h1>
<div id="generate-simulated-data" class="section level2">
<h2>Generate simulated data</h2>
<p>The simulation study relies on a few R packages and python softwares
(e.g. SpaGCN and FICT), which need to be installed beforehand. The
specific packages and all necessary functions to conduct the simulation
study can be found at <a
href="https://github.com/zhengli09/BASS-Analysis/tree/master/code/">simulation</a>.</p>
<pre class="r"><code>source(&quot;code/simu_utils.R&quot;)
# data from STARmap (BZ5) for inferring parameters for splatter
cnts &lt;- readRDS(&quot;data/simu_cnts.RDS&quot;)
info &lt;- readRDS(&quot;data/simu_info.RDS&quot;)
xy &lt;- as.matrix(info[, c(&quot;x&quot;, &quot;y&quot;)])
starmap &lt;- list(cnts = cnts, info = info)

# we perform 50 simulation replicates under each setting and
# the random seeds used for each replicate are generated as below
NREP &lt;- 50
set.seed(0)
seeds &lt;- sample(20201230, NREP)</code></pre>
<p>We generate the simulated data under the baseline simulation setting.
Refer to the <a
href="https://github.com/zhengli09/BASS-Analysis/tree/master/data/params.txt">params</a>
file for a complete list of simulation settings.</p>
<pre class="r"><code># baseline setting of all four scenarios for 
# the single tissue section analysis (L = 1)
scenarios &lt;- 1:4 # simulation scenarios
C &lt;- 4 # number of cell types
R &lt;- 4 # number of spatial domains
J &lt;- 200 # number of genes
de_prop &lt;- 0.2 # proportion of DE genes for each cell type
de_facLoc &lt;- 1.1 # DE gene strength
L &lt;- 1 # number of tissue sections

# illustrate the data at the first replicate
rep &lt;- 1
sim_dats &lt;- lapply(scenarios, function(scenario){
  simu(starmap = starmap, scenario = scenario, C = C, J = J, L = L, 
    batch_facLoc = 0, de_prop = de_prop, de_facLoc = de_facLoc, 
    de_facScale = 0.4, sim_seed = seeds[rep], debug = FALSE)
})</code></pre>
<pre class="r"><code># The simulated data include an L-list of gene expression count matrices,
# an L-list of true cell type label vectors for evaluation purpose, and 
# the random seed used for generating the data
scenario &lt;- 3
sim_dat &lt;- sim_dats[[scenario]]
# gene expression count matrix of the first tissue section under scenario 3
sim_dat[[1]][[1]][1:5, 1:5]</code></pre>
<pre><code>      Cell1 Cell2 Cell3 Cell4 Cell5
Gene1     0     0     0     0     0
Gene2    17     2     6     9     1
Gene3     0     0     0     0     0
Gene4     0     0     0     0    19
Gene5     0     1     3     0    44</code></pre>
</div>
<div id="visualize-the-simulated-data" class="section level2">
<h2>Visualize the simulated data</h2>
<p>We visualize the cell type distributions under scenarios I-IV. You
can refer to <a
href="https://github.com/zhengli09/BASS-Analysis/tree/master/code/viz.R">visualization</a>
for some useful plotting functions or you can write your own code for
plotting.</p>
<pre class="r"><code>library(cowplot)
source(&quot;code/viz.R&quot;)
p &lt;- lapply(scenarios, function(scenario){
  plotClusters(xy, labels = sim_dats[[scenario]][[2]][[1]], 
    title = paste(&quot;Scenario&quot;, c(&quot;I&quot;, &quot;II&quot;, &quot;III&quot;, &quot;IV&quot;)[scenario]))
})
legend &lt;- get_legend(p[[1]] + 
  theme(legend.position = &quot;bottom&quot;) +
  guides(color = guide_legend(title = &quot;Cell type&quot;)))
p &lt;- plot_grid(plotlist = p, ncol = 2)
plot_grid(p, legend, ncol = 1, rel_heights = c(1, 0.1))</code></pre>
<p><img src="figure/simu.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/zhengli09//BASS-analysis/blob/8b169e7cf961f4b59d9fa4803f7ca92946ad80a2/docs/figure/simu.Rmd/unnamed-chunk-4-1.png" target="_blank">8b169e7</a>
</td>
<td>
zhengli09
</td>
<td>
2022-06-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/zhengli09//BASS-analysis/blob/182beebb01b81150712c13602b2446a0c4405c44/docs/figure/simu.Rmd/unnamed-chunk-4-1.png" target="_blank">182beeb</a>
</td>
<td>
zhengli09
</td>
<td>
2022-03-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/zhengli09//BASS-analysis/blob/9a685114f1f29bb007c27088adaf8ce2582c56a2/docs/figure/simu.Rmd/unnamed-chunk-4-1.png" target="_blank">9a68511</a>
</td>
<td>
zhengli09
</td>
<td>
2022-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="running-all-methods" class="section level2">
<h2>Running all methods</h2>
<p>We have wrapped up code for running different methods into different
functions. Those functions take in simulated data and output various
evaluation quantities that include the adjusted random index (ARI),
<span class="math inline">\(F_1\)</span> score, Matthew’s correlation
coefficient (MCC), estimated number of cell types/spatial domains, and
proportions of cells in each cell type/spatial domain. There, ARI, which
was used as a major evaluation metric in our study, measures the
similarity between the estimated cell type/spatial domain labels and the
underlying truth. Functions for running different methods can be found
at <a
href="https://github.com/zhengli09/BASS-Analysis/tree/master/code/">simulation</a>.
We take the simulated data from scenario 3 as an example to illustrate
running different methods.</p>
<pre class="r"><code>scenario &lt;- 3</code></pre>
<div id="run-bass" class="section level3">
<h3>Run BASS</h3>
<pre class="r"><code>BASS_out &lt;- run_BASS(sim_dats[[scenario]], xy, &quot;SW&quot;, 
  beta = 1, C, R, init_method = &quot;kmeans&quot;)</code></pre>
<pre><code>***************************************
  INPUT INFO:
    - Number of tissue sections: 1 
    - Number of cells/spots: 1127 
    - Number of genes: 200 
    - Potts interaction parameter estimation method: SW 
      - Estimate Potts interaction parameter with SW algorithm
  To list all hyper-parameters, Type listAllHyper(BASS_object)
***************************************
***** Log-normalize gene expression data *****
***** Exclude genes with 0 expression *****
***** Reduce data dimension with PCA *****
Post-processing...
done</code></pre>
<p>Note that for cell type clustering, we evaluate three metrics that
include the ARI, <span class="math inline">\(F_1\)</span>-score, and
MCC. The three metrics are evaluated in all cells, cells of major types,
and cells of rare types. In our main simulation study, we focus on the
ARI evaluated in all cells. The other quantities are relevant to the
additional simulation studies where we evaluate the influence of rare
cell types on the performance of all methods. Refer to the simulation
section of our paper for details.</p>
<pre class="r"><code>BASS_out</code></pre>
<pre><code>$c_ari
  all_ari major_ari  rare_ari 
0.9695299 0.9695299       NaN 

$c_F1
   all_F1  major_F1   rare_F1 
0.9882687 0.9882687        NA 

$c_MCC
  all_MCC major_MCC  rare_MCC 
0.9844597 0.9844597        NA 

$C_est
[1] 4

$c_clust_prop
[1] &quot;0.247559893522626,0.176574977817214,0.273291925465839,0.302573203194321&quot;

$z_ari
[1] 0.9218507

$R_est
[1] 4

$z_clust_prop
[1] &quot;0.109139307897072,0.149068322981366,0.42413487133984,0.317657497781721&quot;

$pi_est
          [,1]        [,2]      [,3]      [,4]
[1,] 0.5203252 0.005952381 0.2635983 0.2458101
[2,] 0.2032520 0.488095238 0.0000000 0.2569832
[3,] 0.2764228 0.255952381 0.4832636 0.0000000
[4,] 0.0000000 0.250000000 0.2531381 0.4972067

$mse_pi
[1] 0.04320112

$beta
[1] 1.095881</code></pre>
</div>
<div id="run-hmrf" class="section level3">
<h3>Run HMRF</h3>
<p>Please note that running HMRF requires you to specify a path to
python and will generate intermediate and final results into a specified
folder. Please modify those directories accordingly in the
<code>run_HMRF</code> function if you want to use the function for your
own analysis.</p>
<pre class="r"><code># Note that case and rep parameters below are used to generate 
# a unique temporary directories for running HMRF.
HMRF_out &lt;- run_HMRF(sim_dat, xy, ztrue = info$z, R, case = 3, 
  rep, usePCs = F, dosearchSEgenes = T)</code></pre>
<pre class="r"><code># ARI for spatial domain detection across different
# spatial interaction parameter betas.
unlist(HMRF_out$ari)</code></pre>
<pre><code>      0.0       2.0       4.0       6.0       8.0      10.0      12.0      14.0 
0.1368218 0.1398503 0.1466260 0.1940655 0.2727412 0.3062912 0.3985583 0.4085346 
     16.0      18.0      20.0      22.0      24.0      26.0      28.0      30.0 
0.4867223 0.5020772 0.5316937 0.5230154 0.5064515 0.5168355 0.5475694 0.5565980 
     32.0      34.0      36.0      38.0      40.0      42.0      44.0      46.0 
0.5419703 0.5323911 0.5578577 0.5122298 0.4937736 0.5133692 0.5231834 0.5336728 
     48.0      50.0 
0.5312293 0.5335755 </code></pre>
</div>
<div id="run-bayesspace" class="section level3">
<h3>Run BayesSpace</h3>
<p>Note that BayesSpace cannot identify any neighbors because it was
developed for analyzing ordered lattice structure of spots from ST or
10x Visium platform.</p>
<pre class="r"><code>BayesSpace_out &lt;- run_BayesSpace(sim_dat, xy, ztrue = info$z, R)</code></pre>
<pre><code>Neighbors were identified for 0 out of 1127 spots.</code></pre>
<pre><code>Fitting model...</code></pre>
<pre><code>Calculating labels using iterations 1000 through 10000.</code></pre>
<pre class="r"><code># ARI for spatial domain detection, estimated number of domains,
# and proportions of cells in each domain
BayesSpace_out</code></pre>
<pre><code>$ari
[1] 0.1515157

$R_est
[1] 4

$z_clust_prop
[1] &quot;0.254658385093168,0.177462289263531,0.26885536823425,0.299023957409051&quot;</code></pre>
</div>
<div id="run-spagcn" class="section level3">
<h3>Run SpaGCN</h3>
<p>Because SpaGCN was originally developed with Python, we wrapped up
the python code into a function and have it imported into R such that we
can conveniently analyze the simulated data in R. The python function
can be found at <a
href="https://github.com/zhengli09/BASS-Analysis/tree/master/code/">simulation</a>.</p>
<pre class="r"><code>source_python(&quot;code/run_SpaGCN.py&quot;)
SpaGCN_out &lt;- run_SpaGCN(sim_dat, xy, info$z, R)
# ARI for spatial domain detection, estimated number of domains,
# and proportions of cells in each domain
SpaGCN_out</code></pre>
<pre><code>$ari
[1] 0.2346377

$R_est
[1] 4

$z_clust_prop
[1] &quot;0.215616681455191,0.186335403726708,0.29547471162378,0.302573203194321&quot;</code></pre>
</div>
<div id="run-seurat" class="section level3">
<h3>Run Seurat</h3>
<p>Because Seurat uses a resolution parameter to indirectly determine
the number of clusters, we run Seurat on a range of resolution
parameters and identify the first value that resulted in the desired
number of cell types.</p>
<pre class="r"><code>resolutions &lt;- seq(0.1, 4, by = 0.1)
seu_out &lt;- seu_cluster(sim_dat, C, resolutions)
# ARI, F1-score, and MCC for cell type clustering, 
# number of identified clusters, and the proportion 
# of cells in each cell type cluster.
seu_out</code></pre>
<pre><code>$metric
  all_ari major_ari  rare_ari    all_F1  major_F1   rare_F1   all_MCC major_MCC 
0.9308230 0.9308230       NaN 0.9726765 0.9726765        NA 0.9638171 0.9638171 
 rare_MCC 
       NA 

$C_est
[1] 4

$clust_props
[1] &quot;0.248447204968944,0.173025732031943,0.270629991126886,0.307897071872227&quot;</code></pre>
</div>
<div id="run-sc3" class="section level3">
<h3>Run SC3</h3>
<p>Note that when the number of cells exceeds 5,000, SC3 randomly
selects 5,000 cells for clustering, trains a support vector machine with
the cluster labels obtained from the 5,000 cells, and then predict the
cluster labels of the remaining cells.</p>
<pre class="r"><code># Note that SC3 is slow
sc3_out &lt;- sc3_cluster(sim_dat, C)</code></pre>
<pre><code>Setting SC3 parameters...</code></pre>
<pre><code>Calculating distances between the cells...</code></pre>
<pre><code>Performing transformations and calculating eigenvectors...</code></pre>
<pre><code>Performing k-means clustering...</code></pre>
<pre><code>Calculating consensus matrix...</code></pre>
<pre class="r"><code># ARI, F1-score, and MCC for cell type clustering, 
# number of identified clusters, and the proportion 
# of cells in each cell type cluster.
sc3_out</code></pre>
<pre><code>$metric
  all_ari major_ari  rare_ari    all_F1  major_F1   rare_F1   all_MCC major_MCC 
0.9602983 0.9602983       NaN 0.9845671 0.9845671        NA 0.9795533 0.9795533 
 rare_MCC 
       NA 

$C_est
[1] 4

$clust_props
[1] &quot;0.249334516415262,0.179236912156167,0.267080745341615,0.304347826086957&quot;</code></pre>
</div>
<div id="run-fict" class="section level3">
<h3>Run FICT</h3>
<p>Please install the software according to the <a
href="https://github.com/haotianteng/FICT">FICT github</a> and modify
the paths to the FICT software and a temporary folder accordingly in the
<code>fict_cluster</code> function.</p>
<pre class="r"><code># Note that case and rep parameters below are used to generate 
# a unique temporary directories for running FICT.
fict_out &lt;- fict_cluster(sim_dat, xy, C, case = 3, rep)
# ARI, F1-score, and MCC for cell type clustering, 
# number of identified clusters, and the proportion 
# of cells in each cell type cluster.
fict_out</code></pre>
<pre><code>$metric
  all_ari major_ari  rare_ari    all_F1  major_F1   rare_F1   all_MCC major_MCC 
0.9566095 0.9566095       NaN 0.9823905 0.9823905        NA 0.9768416 0.9768416 
 rare_MCC 
       NA 

$C_est
[1] 4

$clust_props
[1] &quot;0.245785270629991,0.184560780834073,0.267080745341615,0.302573203194321&quot;</code></pre>
</div>
</div>
</div>
<div id="multi-sample-simulation" class="section level1">
<h1>Multi-sample simulation</h1>
<p>In addition to simulating and analyzing a single tissue section, we
also simulated multiple tissue sections and evaluated the performance of
BASS for multi-sample integrative analysis. Specifically, we generated
additional tissue sections based on the same spatial locations of the
1,127 cells in the single tissue section analysis but with slightly
different spatial domain allocations for them, creating varying spatial
domain boundaries between the four cortical layers on different tissue
sections. In each section, we set the spatial domain boundaries to be
vertical, and created new boundaries by horizontally moving the original
boundaries based on a uniform distribution with step size set to be
approximately 10% (= 500) of the tissue width. The resulting new tissue
sections are as follows. <img src="add_tis_sects.png" /></p>
<p>Here, similar to the single-sample analysis, we generate the
simulated data under the baseline simulation setting but for five tissue
sections (<code>L = 5</code>). Refer to the <a
href="https://github.com/zhengli09/BASS-Analysis/tree/master/data/params_mult.txt">params_mult</a>
file for a complete list of simulation settings.</p>
<pre class="r"><code># generate simulated data for five tissue sections
L &lt;- 5
sim_dat &lt;- simu(
  starmap = starmap, scenario = scenario, C = C, J = J, L = L, 
  batch_facLoc = 0, de_prop = de_prop, de_facLoc = de_facLoc, 
  de_facScale = 0.4, sim_seed = seeds[rep], debug = FALSE)
# run BASS
BASS_out &lt;- run_BASS(sim_dat, xy, &quot;SW&quot;, beta = 1, 
  C, R, init_method = &quot;kmeans&quot;)</code></pre>
<pre><code>***************************************
  INPUT INFO:
    - Number of tissue sections: 5 
    - Number of cells/spots: 1127 1127 1127 1127 1127 
    - Number of genes: 200 
    - Potts interaction parameter estimation method: SW 
      - Estimate Potts interaction parameter with SW algorithm
  To list all hyper-parameters, Type listAllHyper(BASS_object)
***************************************
***** Log-normalize gene expression data *****
***** Exclude genes with 0 expression *****
***** Reduce data dimension with PCA *****
***** Correct batch effect with Harmony *****
Post-processing...
done</code></pre>
<pre class="r"><code>BASS_out</code></pre>
<pre><code>$c_ari
  all_ari major_ari  rare_ari 
0.9641914 0.9641914       NaN 

$c_F1
   all_F1  major_F1   rare_F1 
0.9857277 0.9857277        NA 

$c_MCC
  all_MCC major_MCC  rare_MCC 
0.9811072 0.9811072        NA 

$C_est
[1] 4

$c_clust_prop
[1] &quot;0.242945874001775,0.196628216503993,0.246850044365572,0.31357586512866&quot;

$z_ari
[1] 0.9292019

$R_est
[1] 4

$z_clust_prop
[1] &quot;0.111978704525288,0.156699201419698,0.372670807453416,0.358651286601597&quot;

$pi_est
          [,1]        [,2]      [,3]        [,4]
[1,] 0.5182250 0.001132503 0.2538095 0.251360713
[2,] 0.2662441 0.505096263 0.0000000 0.244433449
[3,] 0.2155309 0.234428086 0.4976190 0.001484414
[4,] 0.0000000 0.259343148 0.2485714 0.502721425

$mse_pi
[1] 0.01284252

$beta
[1] 1.106375</code></pre>
</div>
<div id="additional-simulations" class="section level1">
<h1>Additional simulations</h1>
<p>Besides the main simulations above, we also performed additional
simulations to evaluate the influence of various other factors –
including the specified number of cell types and spatial domains, rare
cell types, and a random exclusion of genes – on method performance.
Please check the simulation section of our paper for detailed
description of the simulation design. A list of simulation settings and
code for running the simulation and for plotting can be found <a
href="https://github.com/zhengli09/BASS-Analysis/tree/master/code/add_simu">here</a>.</p>
<div
id="influence-of-the-specified-number-of-cell-types-and-spatial-domains"
class="section level2">
<h2>Influence of the specified number of cell types and spatial
domains</h2>
<pre class="r"><code>sim_dat &lt;- sim_dats[[scenario]]
# over-specify the number of cell types (C)
BASS_overC &lt;- run_BASS(sim_dat, xy, &quot;SW&quot;, beta = 1, 
  C = 10, R = 4, init_method = &quot;kmeans&quot;)</code></pre>
<pre class="r"><code># Note that the proportions of cells in the redundant clusters are 
# small and the spatial domain detection is only slightly affected
BASS_overC</code></pre>
<pre><code>$c_ari
  all_ari major_ari  rare_ari 
0.9333602 0.9333602       NaN 

$c_F1
   all_F1  major_F1   rare_F1 
0.9881693 0.9881693        NA 

$c_MCC
  all_MCC major_MCC  rare_MCC 
0.9842012 0.9842012        NA 

$C_est
[1] 10

$c_clust_prop
[1] &quot;0.237799467613132,0.17036379769299,0.264418811002662,0.299023957409051,0.0115350488021295,0.0062111801242236,0.00443655723158829,0.00266193433895297,0.00266193433895297,0.000887311446317658&quot;

$z_ari
[1] 0.9136783

$R_est
[1] 4

$z_clust_prop
[1] &quot;0.108251996450754,0.147293700088731,0.425022182786158,0.319432120674357&quot;

$pi_est
[1] NA

$mse_pi
[1] NA

$beta
[1] 1.1399</code></pre>
<pre class="r"><code># over-specify the number of spatial domains (R)
BASS_overR &lt;- run_BASS(sim_dat, xy, &quot;SW&quot;, beta = 1, 
  C = 4, R = 10, init_method = &quot;kmeans&quot;)</code></pre>
<pre class="r"><code># Note that the proportions of cells in the redundant domains are 
# small and the cell type clustering is only slightly affected
BASS_overR</code></pre>
<pre><code>$c_ari
  all_ari major_ari  rare_ari 
0.9650832 0.9650832       NaN 

$c_F1
   all_F1  major_F1   rare_F1 
0.9863794 0.9863794        NA 

$c_MCC
  all_MCC major_MCC  rare_MCC 
0.9820171 0.9820171        NA 

$C_est
[1] 4

$c_clust_prop
[1] &quot;0.249334516415262,0.175687666370896,0.273291925465839,0.301685891748004&quot;

$z_ari
[1] 0.9177349

$R_est
[1] 6

$z_clust_prop
[1] &quot;0.109139307897072,0.145519077196096,0.426796805678793,0.316770186335404,0.000887311446317658,0.000887311446317658&quot;

$pi_est
[1] NA

$mse_pi
[1] NA

$beta
[1] 1.366436</code></pre>
</div>
<div id="influence-of-rare-cell-types" class="section level2">
<h2>Influence of rare cell types</h2>
<pre class="r"><code># rare_dist = spatial/random indicates that the rare cell types 
# are either located in specific domains or distributed randomly 
# across the entire tissue section
# C = 10/14 (4 major cell types + 6/10 rare cell types)
# major cell types consist of 70% of cells in the data while
# rare cell types consist of the remaining 30% of cells
sim_dat_rare &lt;- simu(starmap, scenario = 5, rare_dist = &quot;spatial&quot;,
  C = 14, J = 200, L = 1, batch_facLoc = 0, de_prop = 0.2, 
  de_facLoc = 1.1, de_facScale = 0.4, sim_seed = seeds[rep])</code></pre>
<pre class="r"><code>BASS_rare &lt;- run_BASS(sim_dat_rare, xy, &quot;SW&quot;, beta = 1, 
  C = 14, R = 4, init_method = &quot;kmeans&quot;)</code></pre>
<pre class="r"><code>BASS_rare</code></pre>
<pre><code>$c_ari
  all_ari major_ari  rare_ari 
0.8592284 0.9169639 0.4223043 

$c_F1
   all_F1  major_F1   rare_F1 
0.6342784 0.9567075 0.5053068 

$c_MCC
  all_MCC major_MCC  rare_MCC 
0.6357533 0.9481603 0.5107905 

$C_est
[1] 14

$c_clust_prop
[1] &quot;0.16681455190772,0.116237799467613,0.173025732031943,0.22360248447205,0.0656610470275066,0.0372670807453416,0.0230700976042591,0.031055900621118,0.0692102928127773,0.00532386867790594,0.00443655723158829,0.0434782608695652,0.0372670807453416,0.00354924578527063&quot;

$z_ari
[1] 0.9050962

$R_est
[1] 4

$z_clust_prop
[1] &quot;0.108251996450754,0.153504880212955,0.417923691215617,0.320319432120674&quot;

$pi_est
[1] NA

$mse_pi
[1] NA

$beta
[1] 1.12007</code></pre>
</div>
<div id="influence-of-a-random-exclusion-of-genes"
class="section level2">
<h2>Influence of a random exclusion of genes</h2>
<pre class="r"><code># I = 100, 200, or 500 indicates the number of retained genes 
# while the total number of genes is 1000 in the original gene 
# expression matrix
sim_dat_rand_excl &lt;- simu(starmap, scenario = 3, C = 4, 
  J = 1000, I = 200, L = 1, batch_facLoc = 0, de_prop = 0.2, 
  de_facLoc = 1.1, de_facScale = 0.4, sim_seed = seeds[rep])</code></pre>
<pre class="r"><code>BASS_rand_excl &lt;- run_BASS(sim_dat_rand_excl, xy, &quot;SW&quot;, 
  beta = 1, C = 4, R = 4, init_method = &quot;kmeans&quot;)</code></pre>
<pre class="r"><code>BASS_rand_excl</code></pre>
<pre><code>$c_ari
  all_ari major_ari  rare_ari 
0.7009269 0.7009269       NaN 

$c_F1
   all_F1  major_F1   rare_F1 
0.8748878 0.8748878        NA 

$c_MCC
  all_MCC major_MCC  rare_MCC 
0.8352678 0.8352678        NA 

$C_est
[1] 4

$c_clust_prop
[1] &quot;0.247559893522626,0.193433895297249,0.246672582076309,0.312333629103815&quot;

$z_ari
[1] 0.8896475

$R_est
[1] 4

$z_clust_prop
[1] &quot;0.09849157054126,0.163265306122449,0.406388642413487,0.331854480922804&quot;

$pi_est
           [,1]       [,2]        [,3]      [,4]
[1,] 0.49549550 0.02173913 0.270742358 0.2566845
[2,] 0.26126126 0.53260870 0.006550218 0.2352941
[3,] 0.22522523 0.23369565 0.458515284 0.0000000
[4,] 0.01801802 0.21195652 0.264192140 0.5080214

$mse_pi
[1] 0.1623195

$beta
[1] 1.14088</code></pre>
</div>
</div>
<div id="find-all-the-simulation-results" class="section level1">
<h1>Find all the simulation results</h1>
<p>You can find results for all the simulation studies that include the
main simulation study and additional simulation studies <a
href="https://github.com/zhengli09/BASS-Analysis/tree/master/data/simulation_results">here</a>.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.5 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas/libblas.so.3
LAPACK: /usr/lib/x86_64-linux-gnu/libopenblasp-r0.2.20.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] cowplot_1.1.1               CVXR_1.0-10                
 [3] reticulate_1.25             splatter_1.16.1            
 [5] gtools_3.9.2                scater_1.16.2              
 [7] forcats_0.5.0               stringr_1.4.0              
 [9] dplyr_1.0.8                 purrr_0.3.4                
[11] readr_1.3.1                 tidyr_1.1.1                
[13] tibble_3.1.6                ggplot2_3.3.5              
[15] tidyverse_1.3.0             mclust_5.4.9               
[17] GapClust_0.1.0              SC3_1.24.0                 
[19] Seurat_3.2.3                BayesSpace_1.2.1           
[21] SingleCellExperiment_1.14.1 SummarizedExperiment_1.22.0
[23] Biobase_2.48.0              GenomicRanges_1.44.0       
[25] GenomeInfoDb_1.24.2         IRanges_2.26.0             
[27] S4Vectors_0.30.2            BiocGenerics_0.38.0        
[29] MatrixGenerics_1.4.3        matrixStats_0.61.0         
[31] Giotto_1.0.4                BASS_1.1.0                 
[33] GIGrvg_0.5                  workflowr_1.7.0            

loaded via a namespace (and not attached):
  [1] rappdirs_0.3.3            scattermore_0.8          
  [3] maxLik_1.5-2              coda_0.19-4              
  [5] bit64_4.0.2               knitr_1.37               
  [7] irlba_2.3.3               DelayedArray_0.18.0      
  [9] data.table_1.14.2         rpart_4.1.16             
 [11] RCurl_1.98-1.5            doParallel_1.0.15        
 [13] generics_0.1.2            callr_3.7.0              
 [15] RSQLite_2.2.0             combinat_0.0-8           
 [17] RANN_2.6.1                proxy_0.4-26             
 [19] future_1.25.0             bit_4.0.4                
 [21] xml2_1.3.3                lubridate_1.7.9          
 [23] spatstat.data_2.2-0       httpuv_1.5.4             
 [25] assertthat_0.2.1          viridis_0.5.1            
 [27] xfun_0.29                 hms_0.5.3                
 [29] jquerylib_0.1.4           evaluate_0.15            
 [31] promises_1.1.1            DEoptimR_1.0-11          
 [33] fansi_1.0.2               readxl_1.3.1             
 [35] dbplyr_1.4.4              igraph_1.3.1             
 [37] DBI_1.1.1                 htmlwidgets_1.5.1        
 [39] Rmpfr_0.8-7               ellipsis_0.3.2           
 [41] backports_1.2.1           deldir_1.0-6             
 [43] sparseMatrixStats_1.8.0   vctrs_0.3.8              
 [45] here_0.1                  ROCR_1.0-11              
 [47] abind_1.4-5               rflann_1.5               
 [49] withr_2.4.3               cachem_1.0.6             
 [51] robustbase_0.95-0         checkmate_2.1.0          
 [53] sctransform_0.3.3         scran_1.20.1             
 [55] goftest_1.2-3             cluster_2.1.3            
 [57] lazyeval_0.2.2            Rglpk_0.6-4              
 [59] crayon_1.5.0              slam_0.1-50              
 [61] labeling_0.4.2            edgeR_3.38.0             
 [63] pkgconfig_2.0.3           nlme_3.1-157             
 [65] vipor_0.4.5               rlang_1.0.1              
 [67] globals_0.15.0            lifecycle_1.0.1          
 [69] miniUI_0.1.1.1            sandwich_3.0-1           
 [71] BiocFileCache_1.12.0      modelr_0.1.8             
 [73] rsvd_1.0.3                cellranger_1.1.0         
 [75] rprojroot_2.0.2           polyclip_1.10-0          
 [77] lmtest_0.9-40             rngtools_1.5.2           
 [79] Matrix_1.4-1              Rhdf5lib_1.10.1          
 [81] zoo_1.8-10                reprex_0.3.0             
 [83] beeswarm_0.4.0            whisker_0.4              
 [85] ggridges_0.5.3            processx_3.5.2           
 [87] pheatmap_1.0.12           png_0.1-7                
 [89] viridisLite_0.4.0         bitops_1.0-7             
 [91] getPass_0.2-2             KernSmooth_2.23-20       
 [93] blob_1.2.1                DelayedMatrixStats_1.14.3
 [95] doRNG_1.8.2               parallelly_1.31.1        
 [97] lpSolve_5.6.15            beachmat_2.8.1           
 [99] scales_1.1.1              memoise_2.0.1            
[101] magrittr_2.0.2            plyr_1.8.7               
[103] ica_1.0-2                 zlibbioc_1.34.0          
[105] DirichletReg_0.7-1        compiler_4.2.0           
[107] miscTools_0.6-26          dqrng_0.3.0              
[109] RColorBrewer_1.1-2        rrcov_1.7-0              
[111] fitdistrplus_1.1-8        cli_3.2.0                
[113] XVector_0.32.0            listenv_0.8.0            
[115] patchwork_1.1.1           pbapply_1.5-0            
[117] ps_1.6.0                  Formula_1.2-4            
[119] MASS_7.3-57               mgcv_1.8-40              
[121] tidyselect_1.1.2          stringi_1.7.6            
[123] highr_0.9                 yaml_2.3.5               
[125] BiocSingular_1.4.0        locfit_1.5-9.5           
[127] ggrepel_0.9.1             grid_4.2.0               
[129] sass_0.4.1                tools_4.2.0              
[131] future.apply_1.9.0        label.switching_1.8      
[133] rstudioapi_0.13           bluster_1.6.0            
[135] foreach_1.5.0             git2r_0.28.0             
[137] metapod_1.0.0             gridExtra_2.3            
[139] farver_2.1.0              Rtsne_0.16               
[141] digest_0.6.29             pracma_2.2.9             
[143] shiny_1.5.0               Rcpp_1.0.8.3             
[145] broom_0.7.10              scuttle_1.2.1            
[147] harmony_0.1.0             later_1.1.0.1            
[149] RcppAnnoy_0.0.19          WriteXLS_6.4.0           
[151] httr_1.4.2                Rdpack_2.3               
[153] colorspace_2.0-3          rvest_0.3.6              
[155] fs_1.5.2                  tensor_1.5               
[157] splines_4.2.0             uwot_0.1.11              
[159] statmod_1.4.36            spatstat.utils_2.3-1     
[161] xgboost_1.4.1.1           plotly_4.9.2.1           
[163] gmp_0.6-5                 xtable_1.8-4             
[165] jsonlite_1.8.0            spatstat_1.64-1          
[167] R6_2.5.1                  pillar_1.7.0             
[169] htmltools_0.5.2           mime_0.12                
[171] glue_1.6.2                fastmap_1.1.0            
[173] BiocParallel_1.22.0       BiocNeighbors_1.6.0      
[175] class_7.3-20              codetools_0.2-18         
[177] pcaPP_2.0-1               mvtnorm_1.1-3            
[179] utf8_1.2.2                lattice_0.20-45          
[181] bslib_0.3.1               curl_4.3.2               
[183] ggbeeswarm_0.6.0          leiden_0.4.2             
[185] survival_3.3-1            limma_3.52.0             
[187] rmarkdown_2.12.1          munsell_0.5.0            
[189] e1071_1.7-9               rhdf5_2.32.2             
[191] GenomeInfoDbData_1.2.6    iterators_1.0.13         
[193] smfishHmrf_0.1            haven_2.3.1              
[195] reshape2_1.4.4            gtable_0.3.0             
[197] rbibutils_2.2.8          </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
