<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

       
<meta name="author" content="Stefano Cacciatore" />

<meta name="date" content="2024-05-28" />

<title>Simulated data</title>

<script src="site_libs/header-attrs-2.12.1/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

            
        .section-title {
            font-size: 1.5em;
            margin-top: 20px;
            color: #333;
        }
        .code-section {
            background-color: #f4f4f4;
            padding: 10px;
            border-left: 4px solid #333;
            margin: 20px 0;
            position: relative;
        }
        .copy-button {
            position: absolute;
            top: 10px;
            right: 10px;
            padding: 5px 10px;
            background-color: #333;
            color: #fff;
            border: none;
            cursor: pointer;
            font-size: 16px;
        }
        .copy-button:hover {
            background-color: #555;
        }
        pre {
            margin: 0;
            overflow: auto;
        }

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">KODAMA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Introduction</a>
</li>
<li>
  <a href="">Software tutorial</a>
</li>
<li>
  <a href="">Simulation</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Real data analyses
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
   <li>
      <a href="Simulated.html">Similated data</a>
    </li>
    <li>
      <a href="Single cell data.html">Single cell RNA seq data</a>
    </li>
    <li>
      <a href="">DLPFC</a>
    </li>
  </ul>
</li>

      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/zhengli09//BASS-analysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Simulated Data</h1>
<h4 class="author">Stefano cacciatore</h4>
<h4 class="date">2024-05-28</h4>


</div>
<div id="introduction" class="section level1">

    <h1>Introduction</h1>

    <p>
        
        In the context of working with biological spatial data, such as spatial sequencing data, which provide information on the location of biological molecules in a sample, the goal is to identify clusters of molecules grouped in specific areas, thus indicating regions of particular biological activity. For example, in immunology, analyzing the spatial distribution of immune cells within tissues can reveal insights into immune responses and disease mechanisms. Techniques like KODAMA and UMAP are valuable tools for this type of analysis.
    </p>
    <p>
        In the domain of immunology, we could simulate spatial data to study the distribution of immune cells within a tissue. For instance, simulating the localization of T cells and B cells within a lymph node can help us understand how these cells are organized spatially, which is critical for understanding their interactions and functions in immune responses.
    </p>
<div id="Dataset 1" class="section level1">
    <h2 class="section-title">Dataset 1</h2>
    <p>
        The simulated data in this example are generated to demonstrate how to use the KODAMA and UMAP techniques for cluster and spatial domain analysis. We create two distinct regions representing areas rich in T cells and B cells within the lymph node.
    </p>
      <p align="center">
            <img src="https://gist.github.com/assets/168087487/5b73ff0e-ebbc-490d-a027-8f83b4583d1b" alt="">
        </p>
    <p>
        We generate x and y coordinates for the two regions, assigning labels to simulate different cell types (T cells and B cells).
    </p>

    <div class="code-section">
        <button class="copy-button" onclick="copyToClipboard('code1')" title="Copier le code"><i class="fas fa-copy"></i></button>
        <pre id="code1">
set.seed(1234)
x1 = runif(100, min=-1, max=0)
x2 = runif(100, min=0, max=1)
y1 = runif(100)
y2 = runif(100)
x = c(x1, x2)
y = c(y1, y2)
xy = cbind(x, y)

labels = rep(c(TRUE, FALSE), each=100)
ss = sample(100, 5)
labels[ss] = !labels[ss]
region = rep(1:0, each=100)
data = cbind(rnorm(200, mean=labels, sd=0.1), rnorm(200, mean=labels, sd=0.1))
        </pre>
    </div>
 <p align="center">
            <img src="https://gist.github.com/assets/168087487/1ebb8046-d656-4b5b-9265-c515ae4fc650" alt="">
        </p>
        <div id="Visualization with KODAMA and UMAP" class="section level1">
    <h2 class="section-title">Visualization with KODAMA and UMAP</h2>
    <p>
        We use KODAMA and UMAP methods to analyze and visualize the data, applying dimensionality reduction and identifying clusters and spatial domains.
    </p>
    <div class="code-section">
        <button class="copy-button" onclick="copyToClipboard('code2')" title="Copier le code"><i class="fas fa-copy"></i></button>
        <pre id="code2">
color.code = c("#1d79d0aa", "#53ca3eaa")
kk = KODAMA.matrix.parallel(data, spatial=xy, spatial.resolution=0.1, M=100, FUN="PLS", landmarks=100000, splitting=100, f.par.pls=2, n.cores=4)
vis = RunKODAMAvisualization(kk, method="UMAP")
u = umap(data)$layout
par(mfrow=c(1, 3))

plot(x, y, bg=color.code[region+1], pch=21+2*!labels, cex=1+1.5*region, axes=FALSE, main="Position")
axis(1)
axis(2, las=2)
box()

plot(vis, type="n", axes=FALSE, main="KODAMA")
axis(1)
axis(2, las=2)
box()
points(vis[!labels,], bg=color.code[region+1][!labels], pch=21+2, cex=1+1.5*region[!labels])
points(vis[labels,], bg=color.code[region+1][labels], pch=21+2*0, cex=1+1.5*region[labels])

plot(u, type="n", axes=FALSE, xlab="Dimension 1", ylab="Dimension 2", main="UMAP")
axis(1)
axis(2, las=2)
box()
points(u[!labels,], bg=color.code[region+1][!labels], pch=21+2, cex=1+1.5*region[!labels])
points(u[labels,], bg=color.code[region+1][labels], pch=21+2*0, cex=1+1.5*region[labels])
        </pre>
    </div>
        <p align="center">
            <img src="https://gist.github.com/assets/168087487/1a799a59-a370-488b-a8a1-da761d2c86f9" alt="UMAP"">
        </p>
    

        <p align="center">
            <img src="https://gist.github.com/assets/168087487/bd3fd89e-00e7-4912-8efb-aad93d703ae7" alt="KODAMA"">
        </p>
    <p>
        The generated visualizations show the position of the points, as well as the results of the KODAMA and UMAP analyses. The visualizations indicate the molecular positions and identified clusters, providing a better understanding of the spatial structure of the immune cells within the lymph node. This can help identify regions of interest for further analysis, with significant implications for research in immunology and the development of targeted therapies.
    </p>
<div id="Advanced Manipulations and Visualizations" class="section level1">
    <h2 class="section-title">Part Two: Advanced Manipulations and Visualizations</h2>
    <p>
        In this section, we build upon the previous data generation and visualization methods to demonstrate advanced manipulations and visualizations using KODAMA and UMAP techniques.
    </p>
    <p align="center">
        <img src="https://gist.github.com/assets/168087487/43c6a8e1-0aea-4596-b8b8-63142c47bd9f" alt="">
    </p>
    <h3>Data Generation and Setting the Random Seed</h3>
    <p>
        Continuing from the previous section, we set the random seed and generate data for two distinct regions, assigning labels and specific regions to each data point.
    </p>
    <div class="code-section">
        <button class="copy-button" onclick="copyToClipboard('code2')" title="Copier le code"><i class="fas fa-copy"></i></button>
        <pre id="code2">
set.seed(1234)
x1 = runif(100, min=-1, max=0)
x2 = runif(100, min=0, max=1)
y1 = runif(100)
y2 = runif(100)
x = c(x1, x2)
y = c(y1, y2)
xy = cbind(x, y)

labels = rep(c(TRUE, FALSE), each=100)
region = rep(1:0, each=100)

data = cbind(rnorm(200, mean=1+labels, sd=0.1), rnorm(200, mean=1+labels, sd=0.1), rnorm(200, mean=1+labels, sd=0.1), rnorm(200, mean=1+labels, sd=0.1))

ll = length(data)
ss = sample(ll, ll*0.5)
data[ss] = 0

sel = apply(data, 1, function(x) sum(x > 0)) > 2
data = data[sel,]
region = region[sel]
labels = labels[sel]
xy = xy[sel,]

labels = data > 0
labels = labels[,1] + labels[,2]*2 + labels[,3]*4 + labels[,4]*8
labels = as.numeric(as.factor(labels)) + 1
        </pre>
    </div>
    <p align="center">
        <img src="https://gist.github.com/assets/168087487/6cdc6d76-2d22-49ec-bba3-55408e9c2d4b" alt="">
    </p>
    <h3>Visualization with KODAMA and UMAP </h3>
    <p>
      We apply Principal Component Analysis (PCA) on the filtered data and use KODAMA and UMAP methods to analyze and visualize the data, applying dimensionality reduction and identifying clusters and spatial domains.
    </p>
    <div class="code-section">
        <button class="copy-button" onclick="copyToClipboard('code2')" title="Copier le code"><i class="fas fa-copy"></i></button>
        <pre id="code2">
pca = prcomp(scale(data))$x
kk = KODAMA.matrix.parallel(pca, spatial=xy, spatial.resolution=0.1, M=100, FUN="PLS", landmarks=100000, splitting=100, f.par.pls=10, n.cores=4)
config = umap.defaults
config$n_neighbors = 15
vis = RunKODAMAvisualization(kk, method="UMAP", config=config)
u = umap(pca)$layout
pdf("/path/to/save/Figure.pdf", width=7, height=3)
par(mfrow=c(1, 3))

plot(xy, bg=labels, pch=21+2*(region), cex=2+1*region, axes=FALSE, main="Position")
axis(1)
axis(2, las=2)
box()

plot(u, bg=labels, pch=21+2*(region), cex=1+1.5*region, axes=FALSE, main="UMAP")
axis(1)
axis(2, las=2)
box()

dev.off()
  </pre>
</div>
<p align="center">
            <img src="https://gist.github.com/assets/168087487/46c105a5-be8b-44fd-b2b7-03167a870009" alt="UMAP"">
        </p>
        <p align="center">
            <https://gist.github.com/assets/168087487/f47e5c67-0021-4171-8714-4e21c9aacf4f" alt="KODAMA"">
        </p>
<p>
  By integrating advanced manipulations and visualizations, we gain deeper insights into spatial data analysis, providing a foundation for further exploration and research in complex biological systems.
</p>
    </div>
  </div>
</div>
<script>
// Highlight.js initialisation
hljs.initHighlightingOnLoad();
</script>




<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
       
    <script>
        function copyToClipboard(elementId) {
            const codeElement = document.getElementById(elementId);
            const range = document.createRange();
            range.selectNodeContents(codeElement);
            const selection = window.getSelection();
            selection.removeAllRanges();
            selection.addRange(range);
            document.execCommand('copy');
            alert('Code copié dans le presse-papiers');
        }
    </script>

</body>
</html>
