<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Zheng Li" />

<meta name="date" content="2022-06-12" />

<title>Mouse mPFC data by STARmap (Wang et al., 2018)</title>

<script src="site_libs/header-attrs-2.12.1/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">BASS</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Introduction</a>
</li>
<li>
  <a href="about.html">Software tutorial</a>
</li>
<li>
  <a href="simu.html">Simulation</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Real data analyses
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="STARmap.html">STARmap (Wang et al., 2018)</a>
    </li>
    <li>
      <a href="MERFISH.html">MERFISH (Moffitt et al., 2018)</a>
    </li>
    <li>
      <a href="DLPFC.html">DLPFC (Maynard et al., 2021)</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/zhengli09//BASS-analysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Mouse mPFC data by STARmap (Wang et al.,
2018)</h1>
<h4 class="author">Zheng Li</h4>
<h4 class="date">2022-06-12</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-06-12
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>BASS-analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed0code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(0)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed0code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(0)</code> was run prior to running the
code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomzhengli09BASSanalysistree829d10c5addeb411ad5deb4dea9f3ff3e8d79d97targetblank829d10ca">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/zhengli09//BASS-analysis/tree/829d10c5addeb411ad5deb4dea9f3ff3e8d79d97" target="_blank">829d10c</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomzhengli09BASSanalysistree829d10c5addeb411ad5deb4dea9f3ff3e8d79d97targetblank829d10ca"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/zhengli09//BASS-analysis/tree/829d10c5addeb411ad5deb4dea9f3ff3e8d79d97" target="_blank">829d10c</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
working directory clean
</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/STARmap.Rmd</code>) and HTML
(<code>docs/STARmap.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/zhengli09//BASS-analysis/blob/829d10c5addeb411ad5deb4dea9f3ff3e8d79d97/analysis/STARmap.Rmd" target="_blank">829d10c</a>
</td>
<td>
zhengli09
</td>
<td>
2022-06-12
</td>
<td>
Update real data analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/zhengli09//BASS-analysis/67aecc1b77b3c5af9b21c713d8abf11da6dc6954/docs/STARmap.html" target="_blank">67aecc1</a>
</td>
<td>
zhengli09
</td>
<td>
2022-03-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/zhengli09//BASS-analysis/blob/8f34a273f0fb57b1f526da9124ac879aca87974f/analysis/STARmap.Rmd" target="_blank">8f34a27</a>
</td>
<td>
zhengli09
</td>
<td>
2022-03-05
</td>
<td>
Add MERFISH analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/zhengli09//BASS-analysis/79177d12830d8512a21f969ae4cc6abd9a92b8df/docs/STARmap.html" target="_blank">79177d1</a>
</td>
<td>
zhengli09
</td>
<td>
2022-02-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/zhengli09//BASS-analysis/blob/9044081553486614c1ecfa4909655b16758e83d5/analysis/STARmap.Rmd" target="_blank">9044081</a>
</td>
<td>
zhengli09
</td>
<td>
2022-02-27
</td>
<td>
Modify links in STARmap.Rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/zhengli09//BASS-analysis/a27dd42cad44fadeffe07751ac770d859d28a83f/docs/STARmap.html" target="_blank">a27dd42</a>
</td>
<td>
zhengli09
</td>
<td>
2022-02-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/zhengli09//BASS-analysis/blob/ada7b84d38930e62d40ec33e901390ceb4a2b81f/analysis/STARmap.Rmd" target="_blank">ada7b84</a>
</td>
<td>
zhengli09
</td>
<td>
2022-02-27
</td>
<td>
Modify links
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/zhengli09//BASS-analysis/9a685114f1f29bb007c27088adaf8ce2582c56a2/docs/STARmap.html" target="_blank">9a68511</a>
</td>
<td>
zhengli09
</td>
<td>
2022-02-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/zhengli09//BASS-analysis/blob/4ec8df1262087c7c78f682e56014a7087a756d4f/analysis/STARmap.Rmd" target="_blank">4ec8df1</a>
</td>
<td>
zhengli09
</td>
<td>
2022-02-27
</td>
<td>
Publish the introduction, simulation analysis and STARmap analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>Here, we apply BASS to analyze the mouse medial prefrontal cortex
(mPFC) data by the STARmap spatial transcriptomic technology from <a
href="https://www.science.org/doi/10.1126/science.aat5691">Wang et al.,
2018</a>. We focus on the tissue sections BZ5, BZ9 and BZ14 that were
measured on the mPFC region of different mice. The original data can be
downloaded from <a
href="https://www.starmapresources.org/data">here</a>. We excluded cells
that were annotated to be “NA” class by the original study as they were
not confidently identified to be any cell type. Finally, we obtained the
same set of 166 genes measured on 1,049 (BZ5), 1,053 (BZ9), and 1,088
(BZ14) cells along with their centroid coordinates for the downstream
analysis. We manually annotated the three samples with spatial domain
labels, allowing us to quantitatively evaluate different methods with
ARI. There, the manual annotation of spatial domains was based on the
spatial expression of marker genes and the histology diagram of the
mouse brain from the <a
href="https://mouse.brain-map.org/static/atlas">Allen brain atlas</a>.
The processed data and manual annotations can be found at the <a
href="https://github.com/zhengli09/BASS-Analysis/tree/master/data">data
directory</a>. For detailed usage of all the functions, please refer to
the software tutorial section.</p>
<pre class="r"><code>library(BASS)
library(Seurat)
library(tidyverse)
# starmap_cnts, starmap_info
load(&quot;data/starmap_mpfc.RData&quot;)</code></pre>
</div>
<div id="load-data-and-set-hyper-parameters" class="section level1">
<h1>Load data and set hyper-parameters</h1>
<pre class="r"><code>cnts &lt;- starmap_cnts # a list of gene expression count matrices
xys &lt;- lapply(starmap_info, function(info.i){
  as.matrix(info.i[, c(&quot;x&quot;, &quot;y&quot;)])
}) # a list of spatial coordinates matrices
cnts[[&quot;20180417_BZ5_control&quot;]][1:5,1:5]</code></pre>
<pre><code>        69x4486 93x1063 143x3445 88x4092 120x4293
Acss1         0       0        0       0        0
Adcyap1       0       3        0       0        0
Adgrl2        0       0       16       1        7
Aqp4          0       0        8      11       32
Arc           3       3        1       5        0</code></pre>
<pre class="r"><code>head(xys[[&quot;20180417_BZ5_control&quot;]])</code></pre>
<pre><code>                 x        y
69x4486   68.86374 4486.029
93x1063   92.84376 1062.676
143x3445 142.82557 3444.635
88x4092   88.18908 4092.441
120x4293 119.93251 4293.439
104x2648 103.63420 2647.580</code></pre>
<pre class="r"><code># hyper-parameters
C &lt;- 15 # number of cell types
R &lt;- 4 # number of spatial domains</code></pre>
</div>
<div id="run-bass" class="section level1">
<h1>Run BASS</h1>
<pre class="r"><code>set.seed(0)
# Set up BASS object
BASS &lt;- createBASSObject(cnts, xys, C = C, R = R, beta_method = &quot;SW&quot;)</code></pre>
<pre><code>***************************************
  INPUT INFO:
    - Number of tissue sections: 3 
    - Number of cells/spots: 1049 1053 1088 
    - Number of genes: 166 
    - Potts interaction parameter estimation method: SW 
      - Estimate Potts interaction parameter with SW algorithm
  To list all hyper-parameters, Type listAllHyper(BASS_object)
***************************************</code></pre>
<pre class="r"><code># Data pre-processing:
# 1.Library size normalization followed with a log2 transformation
# 2.Dimension reduction with PCA after standardizing all the genes
# 3.Batch effect adjustment using the Harmony package
BASS &lt;- BASS.preprocess(BASS, doLogNormalize = TRUE, 
  doPCA = TRUE, scaleFeature = TRUE, nPC = 20)</code></pre>
<pre><code>***** Log-normalize gene expression data *****
***** Exclude genes with 0 expression *****
***** Reduce data dimension with PCA *****
***** Correct batch effect with Harmony *****</code></pre>
<pre class="r"><code># Run BASS algorithm
BASS &lt;- BASS.run(BASS)</code></pre>
<pre class="r"><code># The spatial parameter beta has converged
# after checking the trace plot of beta
plot(1:BASS@burnin, BASS@samples$beta, xlab = &quot;Iteration&quot;, 
  ylab = expression(beta), type = &quot;l&quot;)</code></pre>
<p><img src="figure/STARmap.Rmd/unnamed-chunk-4-1.png" width="432" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Post-process posterior samples:
# 1.Adjust for label switching with the ECR-1 algorithm
# 2.Summarize the posterior samples to obtain the cell type labels, spatial 
#   domain labels, and the cell type proportion matrix
BASS &lt;- BASS.postprocess(BASS)</code></pre>
<pre><code>Post-processing...
done</code></pre>
<pre class="r"><code>clabels &lt;- BASS@results$c # cell type clusters
zlabels &lt;- BASS@results$z # spatial domain labels
pi_est &lt;- BASS@results$pi # cell type composition matrix</code></pre>
</div>
<div id="annotate-cell-types" class="section level1">
<h1>Annotate cell types</h1>
<pre class="r"><code># Perform DE analysis with Seurat
cnts_all &lt;- do.call(cbind, cnts)
seu_obj &lt;- CreateSeuratObject(counts = cnts_all)
seu_obj &lt;- NormalizeData(seu_obj)
seu_obj &lt;- ScaleData(seu_obj, features = rownames(seu_obj))</code></pre>
<pre><code>Centering and scaling data matrix</code></pre>
<pre class="r"><code>seu_obj &lt;- RunPCA(seu_obj, features = rownames(seu_obj), verbose = F)
Idents(seu_obj) &lt;- factor(unlist(clabels))
markers &lt;- FindAllMarkers(seu_obj, only.pos = T, 
  min.pct = 0, logfc.threshold = 0, verbose = F)
top5 &lt;- markers %&gt;%
  group_by(cluster) %&gt;%
  top_n(n = 5, wt = avg_logFC)

# By checking the top DE genes of each cell type cluster, 
# we annotate specific cell types for each cluster
cTypes &lt;- c(
  &quot;eL5d&quot;, &quot;Astro&quot;, &quot;Micro&quot;, &quot;Lhx6&quot;, &quot;eL6a&quot;,
  &quot;eL2/3&quot;, &quot;eL5c&quot;, &quot;Smc&quot;, &quot;eL6b&quot;, &quot;eL5b&quot;,
  &quot;eL5a&quot;, &quot;Oligo&quot;, &quot;SST&quot;, &quot;Reln&quot;, &quot;VIP&quot;)
clabels &lt;- lapply(clabels, function(clabels.l){
  clabels.l &lt;- factor(clabels.l)
  levels(clabels.l) &lt;- cTypes
  clabels.l &lt;- factor(clabels.l, levels = c(
    &quot;eL2/3&quot;, &quot;eL5a&quot;, &quot;eL5b&quot;, &quot;eL5c&quot;, &quot;eL5d&quot;, 
    &quot;eL6a&quot;, &quot;eL6b&quot;, &quot;Reln&quot;, &quot;VIP&quot;, &quot;SST&quot;,&quot;Lhx6&quot;,
    &quot;Oligo&quot;, &quot;Smc&quot;, &quot;Astro&quot;, &quot;Micro&quot;
  ))
})</code></pre>
<p>
<a class="btn btn-primary" data-toggle="collapse" href="#collapseExample1" role="button" aria-expanded="false" aria-controls="collapseExample1">
Top DE genes for each cell type cluster </a>
</p>
<div id="collapseExample1" class="collapse">
<div class="card card-body">
<pre class="r"><code>data.frame(top5)</code></pre>
<pre><code>           p_val avg_logFC pct.1 pct.2     p_val_adj cluster    gene
1  2.876233e-124 2.3092911 0.761 0.114 4.774546e-122       1    Egr2
2   3.139869e-73 1.6184124 0.957 0.389  5.212183e-71       1    Egr4
3   1.971104e-70 1.8823791 0.920 0.341  3.272033e-68       1     Fos
4   4.449469e-44 1.7215121 0.442 0.094  7.386119e-42       1   Npas4
5   6.150811e-38 1.4558075 0.650 0.231  1.021035e-35       1    Bdnf
6  1.378796e-114 2.5568852 0.694 0.195 2.288802e-112       2    Aqp4
7   7.890228e-52 1.4258715 0.825 0.646  1.309778e-49       2 Pcdhgc3
8   4.200450e-18 1.7168932 0.290 0.129  6.972747e-16       2    Rorb
9   1.772491e-09 1.3918680 0.449 0.371  2.942335e-07       2    Igtp
10  1.386255e-07 1.5058103 0.217 0.124  2.301184e-05       2 Hsd11b1
11  1.023927e-65 2.4703129 0.722 0.184  1.699718e-63       3    Flt1
12  1.252955e-49 3.1147874 0.391 0.061  2.079906e-47       3    Lmo2
13  3.286927e-31 2.1828307 0.429 0.111  5.456298e-29       3   Rgs10
14  4.714697e-29 2.5231186 0.338 0.076  7.826398e-27       3   Itgam
15  8.636498e-21 2.2719544 0.353 0.109  1.433659e-18       3    Car4
16 4.790353e-151 2.2967005 0.975 0.186 7.951986e-149       4    Gad2
17 4.834962e-130 2.7611096 0.981 0.261 8.026037e-128       4    Gad1
18  7.633709e-78 2.3072336 0.540 0.087  1.267196e-75       4    Lhx6
19  5.183467e-63 1.8881946 0.925 0.475  8.604555e-61       4    Rbp4
20  1.780421e-19 1.7123612 0.242 0.060  2.955499e-17       4   Pvalb
21 3.279767e-258 2.2253915 0.747 0.123 5.444414e-256       5    Syt6
22 1.232734e-250 1.6530803 0.983 0.486 2.046339e-248       5    Pcp4
23 6.434915e-206 1.6893945 0.878 0.288 1.068196e-203       5     Sla
24 1.525016e-127 1.2304348 0.592 0.146 2.531526e-125       5   Foxp2
25  2.476512e-43 1.3486708 0.171 0.029  4.111010e-41       5    Rprm
26 5.497035e-167 2.3000464 0.832 0.174 9.125078e-165       6    Cux2
27  1.840535e-80 1.3935963 0.771 0.275  3.055289e-78       6   Cpne5
28  5.933843e-78 1.4504471 0.586 0.152  9.850179e-76       6   Synpr
29  1.782050e-74 1.5709955 0.697 0.243  2.958203e-72       6    Nos1
30  3.374784e-59 1.1418423 0.872 0.524  5.602142e-57       6     Nov
31 1.159966e-111 1.3464488 0.487 0.086 1.925543e-109       7    Tpbg
32  2.515657e-91 1.0283563 0.755 0.283  4.175991e-89       7 Fam19a1
33  3.465477e-87 1.2598938 0.641 0.216  5.752691e-85       7 Tcerg1l
34  5.266243e-65 1.0158491 0.626 0.246  8.741963e-63       7 Adcyap1
35  2.928878e-33 1.1546268 0.470 0.216  4.861937e-31       7    Bdnf
36 1.605416e-155 4.8446465 0.803 0.108 2.664991e-153       8     Mgp
37 8.794806e-135 3.4458575 0.921 0.200 1.459938e-132       8     Bgn
38 3.943231e-127 3.6464290 0.605 0.064 6.545764e-125       8    Mylk
39  7.378080e-61 2.0564594 0.678 0.183  1.224761e-58       8    Flt1
40  3.404510e-12 2.2026007 0.243 0.089  5.651486e-10       8    Spp1
41  2.258126e-87 2.7455940 0.948 0.324  3.748489e-85       9    Ctgf
42  1.486451e-47 1.2324767 0.978 0.570  2.467508e-45       9    Pcp4
43  1.249261e-27 1.4360043 0.739 0.357  2.073774e-25       9   Cplx3
44  2.177792e-26 1.1879363 0.627 0.238  3.615135e-24       9   Nr4a2
45  1.253685e-24 1.9400625 0.194 0.028  2.081118e-22       9   Obox3
46  2.271656e-70 2.0298432 0.958 0.451  3.770948e-68      10    Etv1
47  8.568442e-68 2.0243252 0.909 0.336  1.422361e-65      10    Myl4
48  1.695122e-34 1.6233222 0.587 0.197  2.813903e-32      10  Plcxd2
49  2.621470e-20 1.4652994 0.399 0.132  4.351641e-18      10    Tpbg
50  2.127069e-04 1.0230826 0.105 0.041  3.530935e-02      10  Kcnip2
51  2.043359e-91 1.8364825 0.599 0.133  3.391976e-89      11  Sema3e
52  3.206573e-43 1.2351782 0.533 0.190  5.322912e-41      11  Sema3c
53  9.927517e-39 1.5260025 0.358 0.097  1.647968e-36      11     Npy
54  5.188958e-32 0.9891631 0.697 0.371  8.613670e-30      11   Rspo2
55  1.837275e-14 1.1637204 0.161 0.048  3.049877e-12      11   Tacr1
56  2.119861e-92 2.9567126 0.942 0.273  3.518970e-90      12   Enpp2
57  1.827481e-86 3.1145344 0.777 0.163  3.033618e-84      12     Mog
58  1.616774e-06 0.9739285 0.612 0.514  2.683845e-04      12   Sulf2
59  6.307308e-05 1.1828195 0.124 0.046  1.047013e-02      12  Pdgfra
60  1.016140e-03 1.4340611 0.149 0.074  1.686793e-01      12   Pde1c
61  4.873556e-57 3.3759654 0.844 0.285  8.090102e-55      13     Sst
62  1.924884e-11 1.0333617 0.458 0.184  3.195307e-09      13   Synpr
63  1.776572e-04 0.8851463 0.083 0.023  2.949109e-02      13   Qrfpr
64  2.464206e-03 2.4064675 0.271 0.166  4.090583e-01      13   Calb2
65  3.300751e-03 0.4482856 0.083 0.218  5.479247e-01      13   Nptx2
66 1.710980e-150 4.3247819 0.815 0.039 2.840226e-148      14    Pnoc
67  1.866259e-58 3.2149939 0.981 0.212  3.097989e-56      14    Gad2
68  3.593521e-50 2.8830781 1.000 0.285  5.965245e-48      14    Gad1
69  2.549962e-40 3.3930013 0.648 0.107  4.232936e-38      14    Reln
70  1.061489e-21 2.4232361 0.333 0.047  1.762072e-19      14    Ndnf
71  1.941501e-58 5.2381718 0.838 0.097  3.222891e-56      15     Vip
72  7.585684e-33 2.3914599 0.811 0.161  1.259223e-30      15   Calb2
73  5.231229e-23 1.5710322 0.811 0.185  8.683840e-21      15   Synpr
74  9.134001e-16 1.8958560 0.757 0.259  1.516244e-13      15    Penk
75  9.049567e-14 2.3228765 0.514 0.129  1.502228e-11      15   Htr3a</code></pre>
</div>
</div>
</div>
<div id="visualization" class="section level1">
<h1>Visualization</h1>
<p>You can refer to <a
href="https://github.com/zhengli09/BASS-Analysis/tree/master/code/viz.R">visualization</a>
for some useful plotting functions or you can write your own code for
plotting.</p>
<pre class="r"><code>source(&quot;code/viz.R&quot;)</code></pre>
<div id="spatial-domains" class="section level2">
<h2>Spatial domains</h2>
<pre class="r"><code>zlabels &lt;- lapply(zlabels, function(zlabels.l){
  zlabels.l &lt;- factor(zlabels.l)
  levels(zlabels.l) &lt;- c(&quot;L6&quot;, &quot;L5&quot;, &quot;L2/3&quot;, &quot;L1&quot;)
  zlabels.l &lt;- factor(zlabels.l, levels = c(&quot;L1&quot;, &quot;L2/3&quot;, &quot;L5&quot;, &quot;L6&quot;))
})
cols &lt;-  c(&quot;#e07a5f&quot;, &quot;#3d405b&quot;, &quot;#81b29a&quot;, &quot;#f2cc8f&quot;)
plotClustersFacet(xys, zlabels, c(&quot;BZ5&quot;, &quot;BZ9&quot;, &quot;BZ14&quot;), size = 0.7) +
  scale_color_manual(&quot;Domain&quot;, values = cols)</code></pre>
<p><img src="figure/STARmap.Rmd/unnamed-chunk-9-1.png" width="720" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/zhengli09//BASS-analysis/blob/9a685114f1f29bb007c27088adaf8ce2582c56a2/docs/figure/STARmap.Rmd/unnamed-chunk-9-1.png" target="_blank">9a68511</a>
</td>
<td>
zhengli09
</td>
<td>
2022-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cell-types" class="section level2">
<h2>Cell types</h2>
<pre class="r"><code>cols &lt;- c(&quot;#98c1d9&quot;,&quot;#2a9d8f&quot;,&quot;#e9c46a&quot;,&quot;#f4a261&quot;,&quot;#e76f51&quot;,
  &quot;#234888&quot;, &quot;#a8dadc&quot;,&quot;#b388eb&quot;,&quot;#bb3e03&quot;,&quot;#52b69a&quot;,
  &quot;#e9d8a6&quot;, &quot;#f19c79&quot;, &quot;#DBC9D8&quot;, &quot;#A4277C&quot;, &quot;#D79739&quot;)
plotClustersFacet(xys, clabels, c(&quot;BZ5&quot;, &quot;BZ9&quot;, &quot;BZ14&quot;), size = 0.5) +
  scale_color_manual(&quot;Type&quot;, values = cols) +
  guides(color = guide_legend(byrow = T, nrow = 2,
    override.aes = list(size = 2)))</code></pre>
<p><img src="figure/STARmap.Rmd/unnamed-chunk-10-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="spatial-distribution-of-cell-types-on-the-tissue-section-bz5"
class="section level2">
<h2>Spatial distribution of cell types on the tissue section BZ5</h2>
<pre class="r"><code>library(gghighlight)
plotCellTypes(xys[[1]], clabels[[1]], cols, ncol = 5, dotsize = 0.7,
  size = 0.5, alpha = 0.5)</code></pre>
<p><img src="figure/STARmap.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="cell-type-compositions-in-each-spatial-domain"
class="section level2">
<h2>Cell type compositions in each spatial domain</h2>
<pre class="r"><code># adjust order of labeling
pi_est &lt;- pi_est[match(
  levels(clabels[[1]]), cTypes),
  c(4,3,2,1)]
colnames(pi_est) &lt;- c(&quot;L1&quot;, &quot;L2/3&quot;, &quot;L5&quot;, &quot;L6&quot;)
rownames(pi_est) &lt;- levels(clabels[[1]])
plotCellTypeCompBar(pi_est, cols, nrow = 2)</code></pre>
<p><img src="figure/STARmap.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.5 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas/libblas.so.3
LAPACK: /usr/lib/x86_64-linux-gnu/libopenblasp-r0.2.20.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] gghighlight_0.3.2 forcats_0.5.0     stringr_1.4.0     dplyr_1.0.8      
 [5] purrr_0.3.4       readr_1.3.1       tidyr_1.1.1       tibble_3.1.6     
 [9] ggplot2_3.3.5     tidyverse_1.3.0   Seurat_3.2.3      BASS_1.1.0       
[13] GIGrvg_0.5        workflowr_1.7.0  

loaded via a namespace (and not attached):
  [1] utf8_1.2.2                  reticulate_1.25            
  [3] tidyselect_1.1.2            htmlwidgets_1.5.1          
  [5] combinat_0.0-8              grid_4.2.0                 
  [7] BiocParallel_1.22.0         lpSolve_5.6.15             
  [9] Rtsne_0.16                  munsell_0.5.0              
 [11] codetools_0.2-18            ica_1.0-2                  
 [13] future_1.25.0               miniUI_0.1.1.1             
 [15] withr_2.4.3                 colorspace_2.0-3           
 [17] Biobase_2.48.0              highr_0.9                  
 [19] knitr_1.37                  rstudioapi_0.13            
 [21] stats4_4.2.0                SingleCellExperiment_1.14.1
 [23] ROCR_1.0-11                 tensor_1.5                 
 [25] listenv_0.8.0               labeling_0.4.2             
 [27] MatrixGenerics_1.4.3        git2r_0.28.0               
 [29] GenomeInfoDbData_1.2.6      harmony_0.1.0              
 [31] polyclip_1.10-0             farver_2.1.0               
 [33] rprojroot_2.0.2             parallelly_1.31.1          
 [35] vctrs_0.3.8                 generics_0.1.2             
 [37] xfun_0.29                   R6_2.5.1                   
 [39] GenomeInfoDb_1.24.2         ggbeeswarm_0.6.0           
 [41] rsvd_1.0.3                  bitops_1.0-7               
 [43] spatstat.utils_2.3-1        DelayedArray_0.18.0        
 [45] assertthat_0.2.1            promises_1.1.1             
 [47] scales_1.1.1                beeswarm_0.4.0             
 [49] gtable_0.3.0                globals_0.15.0             
 [51] processx_3.5.2              goftest_1.2-3              
 [53] rlang_1.0.1                 splines_4.2.0              
 [55] lazyeval_0.2.2              broom_0.7.10               
 [57] yaml_2.3.5                  reshape2_1.4.4             
 [59] abind_1.4-5                 modelr_0.1.8               
 [61] backports_1.2.1             httpuv_1.5.4               
 [63] tools_4.2.0                 ellipsis_0.3.2             
 [65] jquerylib_0.1.4             RColorBrewer_1.1-2         
 [67] BiocGenerics_0.38.0         ggridges_0.5.3             
 [69] Rcpp_1.0.8.3                plyr_1.8.7                 
 [71] sparseMatrixStats_1.8.0     zlibbioc_1.34.0            
 [73] RCurl_1.98-1.5              ps_1.6.0                   
 [75] rpart_4.1.16                deldir_1.0-6               
 [77] pbapply_1.5-0               viridis_0.5.1              
 [79] cowplot_1.1.1               S4Vectors_0.30.2           
 [81] zoo_1.8-10                  SummarizedExperiment_1.22.0
 [83] haven_2.3.1                 ggrepel_0.9.1              
 [85] cluster_2.1.3               fs_1.5.2                   
 [87] magrittr_2.0.2              data.table_1.14.2          
 [89] scattermore_0.8             lmtest_0.9-40              
 [91] reprex_0.3.0                RANN_2.6.1                 
 [93] whisker_0.4                 fitdistrplus_1.1-8         
 [95] matrixStats_0.61.0          hms_0.5.3                  
 [97] patchwork_1.1.1             mime_0.12                  
 [99] evaluate_0.15               xtable_1.8-4               
[101] mclust_5.4.9                readxl_1.3.1               
[103] IRanges_2.26.0              gridExtra_2.3              
[105] compiler_4.2.0              scater_1.16.2              
[107] KernSmooth_2.23-20          crayon_1.5.0               
[109] htmltools_0.5.2             mgcv_1.8-40                
[111] later_1.1.0.1               lubridate_1.7.9            
[113] DBI_1.1.1                   dbplyr_1.4.4               
[115] MASS_7.3-57                 Matrix_1.4-1               
[117] cli_3.2.0                   parallel_4.2.0             
[119] igraph_1.3.1                GenomicRanges_1.44.0       
[121] pkgconfig_2.0.3             getPass_0.2-2              
[123] plotly_4.9.2.1              xml2_1.3.3                 
[125] vipor_0.4.5                 bslib_0.3.1                
[127] XVector_0.32.0              rvest_0.3.6                
[129] callr_3.7.0                 digest_0.6.29              
[131] sctransform_0.3.3           RcppAnnoy_0.0.19           
[133] spatstat.data_2.2-0         rmarkdown_2.12.1           
[135] cellranger_1.1.0            leiden_0.4.2               
[137] uwot_0.1.11                 DelayedMatrixStats_1.14.3  
[139] shiny_1.5.0                 lifecycle_1.0.1            
[141] nlme_3.1-157                jsonlite_1.8.0             
[143] BiocNeighbors_1.6.0         limma_3.52.0               
[145] viridisLite_0.4.0           fansi_1.0.2                
[147] pillar_1.7.0                lattice_0.20-45            
[149] fastmap_1.1.0               httr_1.4.2                 
[151] survival_3.3-1              glue_1.6.2                 
[153] spatstat_1.64-1             label.switching_1.8        
[155] png_0.1-7                   stringi_1.7.6              
[157] sass_0.4.1                  blob_1.2.1                 
[159] BiocSingular_1.4.0          irlba_2.3.3                
[161] future.apply_1.9.0         </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
